lib/rubocop/cop/style/case_equality.rb

```diff
@@ -33,11 +33,14 @@ class CaseEquality < Base
         extend AutoCorrector
 
         MSG = 'Avoid the use of the case equality operator `===`.'
+        CAMEL_CASE = /[A-Z]+[a-z]+/.freeze
```

Comment:
I think this pattern should be more specific, otherwise `SomeNamespace::REGXP` would pass when it shouldn't.  I'd suggest something like `\A(?=.*[a-z])(?:[A-Z][a-z]*|::)+\z` instead, but I'm not sure  if that's good enough.

Reply:
This regular expression is based on the following: 
https://github.com/rubocop-hq/rubocop/blob/v0.89.1/lib/rubocop/cop/style/inverse_methods.rb#L45

But the suggested regexp seems more better. I updated this PR. Thank you.

Reply:
Considering that the last component of a constant name is all that matters, this pattern could work better: `(\A|::)(?=.*[a-z])([A-Z][a-z0-9]*)+\z`. (I also took digits into account)

In any case, class names like `URI::HTTP` and `Addressable::URI` cannot be handled nicely unless you keep a known class name list.

Reply:
Just do `lhs.children.last.match? /[[:lower:]]/`. The last child is the name of the constant without its scope.

Reply:
Maybe I should introduce a `ConstNode` and provide `class_const?`

Reply:
How about this... rubocop-hq/rubocop-ast#99

Reply:
@marcandre Thank you for creating the new API. I'd like to replace it with `class_name?` when new RuboCop AST is released.

Reply:
@koic I added it as `module_name?` and aliased it to `class_name?` 👍

Reply:
I've updated this PR with `module_name?`. This PR requires the next RuboCop AST release (0.4.0?) due to its dependency on the unreleased API.